### PR TITLE
[DOC] Fix of description for bulk request

### PR DESCRIPTION
**Pull request by MetinArslanturk**
Detailed description added for client libraries like NodeJs.

See : https://github.com/elastic/elasticsearch-js/pull/1040

### CODE REVIEW

**Comment by MetinArslanturk** [comment]
@lcawl can you label and check this please? The elasticsearch-js api reference is auto-generate from this source and bulk request has huge misunderstanding. I think this can help to fix it.

**Comment by dliappis** [comment]
@delvedor and/or @Mpdreamz could you please take a look at this?

File: rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json [hunk 1]
```diff
@@ -90,7 +90,8 @@
       }
     },
     "body":{
-      "description":"The operation definition and data (action-data pairs), separated by newlines",
+      "type":"list",
+      "description":"The operation definition and data (action-data pairs), separated by newlines E.g. 'action\ndata\naction\naction\ndata'. In client libraries (like NodeJS) action-data pair objects should be used as array members; data is optional according to action type. E.g. '[action,data,action,action,data]'",
```
Comment (dliappis) [inline rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json]: IMHO a lot of the clarifications here are already explained in the reference documentation for bulk: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html (e.g. NDJSON structure etc.)

**Review by dliappis — COMMENTED** [review]
Thank you for your interest. I left a comment, but I'll defer to @delvedor for more comments.

File: rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json (hunk 1 above)
Comment (Mpdreamz) [reply rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json]: I concur with with @dliappis, the description is serves as short description what gets send not necessarily documenting how.

Comment (MetinArslanturk) [reply rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json]: It is okay but if you check this: https://github.com/elastic/elasticsearch-js/pull/1040 the elasticsearch-js reference is auto-generated from these descriptions. In api reference bulk request's body argument has explanation like "seperated by new lines json". This is meanless for NodeJS. So i created a PR for that repo but @delvedor point it, that is an auto-generated file. So i think there are things to fix in elasticsearch-js api reference. Bulk request body argument type called object in there and i saw lots of complications about this on web. This is just clarification.

Comment (delvedor) [reply rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json]: I agree with @dliappis and @Mpdreamz. This description should be short and sweet, a link to the official documentation is provided as well.
Probably what we can do in the js client is to add another link to the [examples](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/bulk_examples.html) section.

**Pull request closed without merging**